### ISSUE TITLE
Map closing without saving when using other app in Android phone

### ISSUE THREAD

**Issue by Zerbembasqwibo** [OpenOrienteering/mapper#968]
### Steps to reproduce
1. Draw anything on a map in Android phone
2. Do not save
3. Switch to camera, or maybe also other apps
4. Close that app
5. Return to OOM

### Actual behaviour
When you come back the map is closed without saving last changes. OOM is not closed.
N.B. It did not ALWAYS happen, but approx 50% of the times. Also some other app did same result (but at that time I learnt to save before!)

### Expected behaviour
Map should not be closed using other apps in phone

### Configuration
Mapper Version: 0.7.91
Operating System: Android 6.0.1 on Fairphone 2

**Comment by yevhenmazur** [comment]
I can confirm this issue.
The issue has started to reproduce after update to Android 6.0 on my phone. It affects also Android up to version 7.1.1 and Mapper from version 0.6.8 (the older versions have not tested).

**Comment by dg0yt** [comment]
Of course, it is more likely to happen after significant Android updates.

Android reserves the right to terminate apps which are in the background. So far, we don't save automatically when Mapper goes to the background. One reason is the fact that saving is an explicit user action at the moment. Another reason is that saving might take more time than Android will give us in that moment, so that Mapper will be forcefully terminated in the middle of saving, anyway.

Things which mitigate the risk of losing changes:
- Autosave when going to background. Easy, but must never destroy a complete autosave file when killed during save.
- Limit the amount of data to be saved, e.g. only record changes. Hard.
